contracts/token/psp34/psp34.rs

```diff
@@ -287,28 +312,31 @@ impl<T: PSP34Storage + Flush> PSP34Internal for T {
         Ok(())
     }
 
-    default fn _mint(&mut self, id: Id) -> Result<(), PSP34Error> {
-        let to = Self::env().caller();
-        self._mint_to(to, id)
-    }
-
     default fn _mint_to(&mut self, to: AccountId, id: Id) -> Result<(), PSP34Error> {
         if self.get_mut().token_owner.get(&id).is_some() {
             return Err(PSP34Error::TokenExists)
         }
 
+        self._before_token_transfer(None, Some(&to), &id)?;
+
         self._add_token(to, id.clone())?;
-        self._emit_transfer_event(None, Some(to), id);
+        self._emit_transfer_event(None, Some(to), id.clone());
+
+        self._after_token_transfer(None, Some(&to), &id)?;
         Ok(())
     }
 
     default fn _burn_from(&mut self, from: AccountId, id: Id) -> Result<(), PSP34Error> {
+        if self.get_mut().token_owner.get(&id).is_none() {
```

Comment:
Let's move this to function `_check_token_exists` and reuse it in `_transfer_token_from`. 

```
 default fn _check_token_exists(&self, id: &Id) -> Result<&AccountId, PSP34Error> {
        self.get().token_owner.get(id).ok_or(PSP34Error::TokenNotExists)
    }
```